src/lib/redux/store.ts

```diff
@@ -0,0 +1,41 @@
+import { PreloadedState, configureStore } from "@reduxjs/toolkit";
+import { listenerMiddleware } from "./listenerMiddleware";
+import { RootState, rootReducer } from "./reducer";
+
+export const createStore = ({
+    preloadedState,
+}: {
+    preloadedState?: PreloadedState<RootState>;
+    // dispatchEvent: (event: CustomEvent) => void;
+}) => {
+    return configureStore({
+        reducer: rootReducer,
+        middleware: (getDefaultMiddleware) =>
+            getDefaultMiddleware({
+                thunk: {
+                    extraArgument: {},
+                },
+                serializableCheck: false,
+            }).prepend(listenerMiddleware.middleware),
+        preloadedState,
+    });
+};
+
+export const store = createStore({});
+export type Store = typeof store;
+
+export const observeStore = <T>(store: Store, select: (state: RootState) => T, onChange: (result: T) => void) => {
+    let currentState: T;
+
+    function handleChange() {
+        const nextState = select(store.getState());
+        if (nextState !== currentState) {
```

Comment:
ah yeah you're right, would probably need a compare function here as well then

Reply:
thinking about it some more, since we're using `createSelector` to create the `select` function, I _think_ the result is memo'd and so this comparison will work... but I'd have to double check

If that is the case, we'd probably want to enforce using a `createSelector` selector here with types

Reply:
it works!
![image](https://github.com/whereby/browser-sdk/assets/138459303/1e44a3d6-9f5c-42e5-bfeb-3fdefeb2a1c9)
```typescript
export const observeStore = <T>(store: Store, select: (state: RootState) => T, onChange: (result: T) => void) => {
    let currentState: T;

    function handleChange() {
        const nextState = select(store.getState());
        console.log("trace state equality", nextState === currentState);
        if (nextState !== currentState) {
            currentState = nextState;
            onChange(currentState);
        }
    }

    const unsubscribe = store.subscribe(handleChange);
    handleChange();
    return unsubscribe;
};
```